torch/_refs/nn/functional/__init__.py

```diff
@@ -610,14 +609,13 @@ def hinge_embedding_loss(
     margin: float = 1.0,
     reduction: str = "mean",
 ) -> TensorLikeType:
-    # Formula of loss (implementation gets confusing with all the refs.foo)
     # loss_without_reduction = input if y == 1
     #                        = max(0, margin - input) if y == -1
     _check_reduction_value(reduction)
-    margin_clamp = refs.maximum(refs.sub(margin, input), 0)
-    output_margin = refs.where(refs.ne(target, 1), margin_clamp, 0)
-    output_self = refs.where(refs.ne(target, -1), input, 0)
-    loss = refs.add(output_margin, output_self)
+    margin_clamp = torch.clamp_min(margin - input, 0)
+    output_margin = torch.where(target != 1, margin_clamp, 0)
+    output_self = torch.where(target != -1, input, 0)
+    loss = output_margin + output_self
```

Comment:
This can be golfed as
```suggestion
    output_margin = torch.where(target == -1, margin_clamp, 0)
    loss = torch.where(target == 1, input, margin_clamp)
```
but this is also an issue in master... where it doesn't even use `0` but creates a full tensor of zeros! my my.

No need to act on this.